ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/AuthenticationPluginHandler.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.auth;
+
+import io.confluent.ksql.api.server.ErrorCodes;
+import io.confluent.ksql.api.server.KsqlApiException;
+import io.confluent.ksql.api.server.Server;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.AuthProvider;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.web.RoutingContext;
+import java.security.Principal;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Handler that calls any authentication plugin
+ */
+public class AuthenticationPluginHandler implements Handler<RoutingContext> {
+
+  private final Server server;
+  private final AuthenticationPlugin securityHandlerPlugin;
+
+  public AuthenticationPluginHandler(final Server server,
+      final AuthenticationPlugin securityHandlerPlugin) {
+    this.server = server;
+    this.securityHandlerPlugin = securityHandlerPlugin;
```

Comment:
nit: validate params that will be stored in object state; ensuring object does not get into an invalid state.

Reply:
ack